cpp/include/cudf/utilities/traits.hpp

```diff
@@ -63,6 +63,18 @@ template <typename L, typename R>
 struct is_equality_comparable_impl<L, R, void_t<equality_comparable<L, R>>> : std::true_type {
 };
 
+template <class From, class To>
+struct is_explicitly_convertible {
```

Comment:
Since these aren't standard traits, you can just make them `constexpr` functions.